src/thin_client.rs

```diff
@@ -156,6 +156,17 @@ impl ThinClient {
                 }
                 self.process_response(resp);
             }
+            match self.recv_response() {
```

Comment:
Did you mean to delete that `if let Ok` above?

Reply:
Ah crap.  Yeah, rebase error I think.  Thanks for catching